torch/ao/quantization/quantizer/qnnpack_quantizer.py

```diff
@@ -0,0 +1,145 @@
+from torch.ao.quantization.backend._qnnpack_pt2e import get_qnnpack_pt2e_backend_config
+
+from .quantizer import Quantizer
+
+import typing
+import copy
+from dataclasses import asdict, dataclass
+from typing import Callable, Tuple, Union
+
+import torch
+
+SUPPORTED_DTYPES = [torch.uint8, torch.int8, torch.int32]
+SUPPORTED_QSCHEMES = [
+    torch.per_tensor_affine,
+    torch.per_channel_affine,
+    torch.per_channel_symmetric,
+    torch.per_channel_affine_float_qparams,
+]
+
+
+@dataclass
+class TensorSpec:
+    dtype: torch.dtype
+    quant_min: int
+    quant_max: int
+    qscheme: torch.per_tensor_affine
+    ch_axis: int
+    is_dynamic: bool = False
+
+    def __post_init__(self):
+        # check dtype is one of the supported types
+        if self.dtype not in SUPPORTED_DTYPES:
+            raise TypeError(f"Unsupported dtype {self.dtype}.")
+
+        # quant_min must be less than quant_max
+        if self.quant_min > self.quant_max:
+            raise ValueError(
+                f"quant_min {self.quant_min} must be <= quant_max {self.quant_max}."
+            )
+
+        # check qscheme is on of the supported ones
+        if self.qscheme not in SUPPORTED_QSCHEMES:
+            raise ValueError(f"Unsupported qscheme {self.qscheme}.")
+
+        # ch_axis must be less than the number of channels
+        # but no way to check here. Just check that it is not < 0.
+        if self.ch_axis < 0:
+            raise ValueError("Ch_axis is < 0.")
+
+
+OperatorSpec = typing.NamedTuple(
+    "OperatorSpec", [("activation", TensorSpec), ("weight", TensorSpec), ("bias", TensorSpec)]
+)
+
+SpecAndOperators = typing.NamedTuple(
+    "SupportedSpecAndOperators",
+    ("operator_spec", OperatorSpec),
+    ("operators", Union[torch.nn.Module, Callable]),
+)
+
+
+def supported_symmetric_quantized_operators():
```

Comment:
I dont know why are these free methods. THis is exactly why I wanted them to be on Quantizer. I think its fine for this prototype but I am gonna add those to quantizer.

Reply:
I think we can discuss, I'd like to start with qnnpack for now, we can decide before we advertise this to other backends if we want to put those in quantizer or not